runner/src/rpc/tool.rs

```diff
@@ -96,3 +107,18 @@ pub(crate) fn credentials_filepath(account_id: AccountId) -> Result<PathBuf, Str
     path.push(format!("{}.json", account_id));
     Ok(path)
 }
+
+/// Convert `StateItem`s over to a Map<data_key, value_bytes> representation.
+/// Assumes key and value are base64 encoded, so this also decode them.
+pub(crate) fn into_state_map(state_items: Vec<StateItem>) -> HashMap<String, Vec<u8>> {
+    let decode = |s: StateItem| {
+        (
+            str::from_utf8(&base64::decode(s.key.clone()).unwrap())
+                .unwrap_or_else(|_| &s.key)
```

Comment:
Wait, you want to use the key if the base64 decoded bytes aren't utf-8? This is all really confusing, if this is the case, can you document the specifics around why these specific semantics are required?

Reply:
Originally when I was playing around with it, the keys looked weird and I couldn't seem to decode any of them, which is why there was this hackish mess. Thanks for pointing it out. I changed it to just return an `anyhow` result instead